testsuite/ui/widgets/__init__.py

```diff
@@ -17,52 +17,91 @@
+    Radio group of 3scale pages.
     """
-    OPTIONS_SECTION = './fieldset/ol[contains(@class, "{}")]'
+    OPTIONS_SECTION_CLASS = './fieldset/ol[contains(@class, "{}")]'
+    OPTIONS_SECTION = './fieldset/ol'
 
     OPTIONS = './li'
-    OPTIONS_BY_ID = OPTIONS + '/label/input[@id="{}"]'
+    OPTIONS_INPUT = OPTIONS + '/label/input'
+    OPTIONS_BY_ID = OPTIONS_INPUT + '[@id="{}"]'
 
     def __init__(self, parent=None, locator=None, fieldset_id="", logger=None):
         super().__init__(parent, locator, logger)
         if fieldset_id:
-            self.options_section = self.browser.element(self.OPTIONS_SECTION.format(fieldset_id))
+            self.options_section = self.browser.element(self.OPTIONS_SECTION_CLASS.format(fieldset_id))
         else:
-            self.options_section = self.browser.element(locator)
+            self.options_section = self.browser.element(self.OPTIONS_SECTION)
 
-    def select(self, options):
+    def select(self, option: str):
         """
         Select radio (check box) element from the list.
-        :param options: String id-s of options
+        :param option: String id-s of options
+        """
+        element = self.browser.element(self.OPTIONS_BY_ID.format(option), parent=self.options_section)
+        element.click()
+
+
+# pylint: disable=abstract-method
+# Widget contains fill method which raise not implemented exception if widget is not fillable but pylint detect it as
+# an abstract method
+class CheckBoxGroup(GenericLocatorWidget):
+    """
+    CheckBox group of 3scale pages
+    :param ol_class: Set if Checkbox group identifier is in ol level of group.
+                     e.g. fieldse/ol[class=ServiceList]
+    :param field_set_name: Set if Checkbox group identifier is in fieldset level of group.
+    """
+    OPTIONS_SECTION = './fieldset/ol'
+    OPTIONS_SECTION_OL_CLASS = './/fieldset/ol[contains(@class, "{}")]'
+    OPTIONS_SECTION_FIELDSET_NAME = './/fieldset[contains(@name, "{}")]/ol'
+
+    OPTIONS = './li'
+    OPTIONS_INPUT = OPTIONS + '/label/input'
+    OPTIONS_BY_ID = OPTIONS_INPUT + '[@id="{}"]'
+
+    # pylint: disable=too-many-arguments
+    def __init__(self, parent=None, locator=None, ol_class=None, field_set_name=None, logger=None):
+        super().__init__(parent, locator, logger)
+        if ol_class:
+            self.options_section = self.browser.element(self.OPTIONS_SECTION_OL_CLASS.format(ol_class))
+        elif field_set_name:
+            self.options_section = self.browser.element(self.OPTIONS_SECTION_FIELDSET_NAME.format(field_set_name))
+        else:
+            self.options_section = self.browser.element(self.OPTIONS_SECTION)
+
+    def is_checked(self, option: str):
+        """Detect if checkbox is already checked"""
+        element = self.browser.element(self.OPTIONS_BY_ID.format(option), parent=self.options_section)
+        return element.is_selected()
+
+    def check(self, options: list):
```

Comment:
this method should be consistent. It should uncheck all options before checking new ones

Reply:
@jsmolar  what about the moment when I want to select an additional option? Here is an example of what I mean:
```python
(pseudocode)
check(["blue", "red"])                        # adds the whole list of options
// do some asserts
check("green")                                # is not supported rn
check(["blue", "red"] + ["green"])           # what i have to do right now
```
The current implementation requires me to send the whole list of options that I want to have checked, but what if I want to keep the old options and just check a single check ?